BUILDING.md

```diff
@@ -162,3 +166,99 @@ publishing {
         ...
 ```
 Use `./gradlew publishShadowPublicationToStagingRepository` to produce maven artifacts. When building as part of the monolithic distribution customize `scripts/build.sh` to collect maven artifacts. See [job-scheduler#71](https://github.com/opensearch-project/job-scheduler/pull/82/files) for an example.
+
+## Custom Gradle Plugins
+
+Automations, Testing other OpenSearch funtionalities can be acheived using Custom Gradle Plugins, there are several custom gradle plugins developed depending on desired use case. The custom gradle plugins are added in [OpenSearch](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc) repo.
+
+### opensearch.pluginzip
+
+This plugin will identify the generated zip file distribution which is the ouput of `bundlePlugin` task and publish to local maven repo with right maven coordinates.
+[Plugin Code](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/main/java/org/opensearch/gradle/pluginzip), [Plugin Tests](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/test/java/org/opensearch/gradle/pluginzip), [Plugin META-INF](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/META-INF/gradle-plugins/opensearch.pluginzip.properties)
+
+
+#### Plugin Design
```

Comment:
I would remove this title.

Reply:
Wont it be easy to check the plugin code location, if a user wants to know more details about this ?

Reply:
The documentation for the plugin belongs with the plugin in its code. You can add these things as comments in the plugin itself. This doc is just about how to use it.